crates/uroborosql-fmt/src/cst/expr/function.rs

```diff
@@ -1,25 +1,193 @@
+        }
+    }
+
+    /// 列リストを複数行で描画するかを指定する。
+    /// true を与えたら必ず複数行で描画され、false を与えたらできるだけ単一行で描画する。
+    pub(crate) fn set_force_multi_line(&mut self, b: bool) {
+        self.force_multi_line = b
+    }
+
+    /// 複数行で描画するかどうかを bool 型の値で取得する。
+    /// 複数行で描画する場合は true を返す。
+    /// 自身の is_multi_line のオプションの値だけでなく、各列が単一行かどうか、末尾コメントを持つかどうかも考慮する。
+    pub(crate) fn is_multi_line(&self) -> bool {
+        self.force_multi_line
+            || self.all_distinct.is_some()
+            || self.order_by.is_some()
+            || self
+                .exprs
+                .iter()
+                .any(|a| a.is_multi_line() || a.has_trailing_comment())
+    }
+
+    /// カラムリストをrenderする。
+    /// 自身の is_multi_line() が true になる場合には複数行で描画し、false になる場合単一行で描画する。
+    pub(crate) fn render(&self, depth: usize) -> Result<String, UroboroSQLFmtError> {
+        // depth は開きかっこを描画する行のインデントの深さ
+        let mut result = String::new();
+
+        if self.is_multi_line() {
+            // 各列を複数行に出力する
+            result.push_str("(\n");
+
+            // ALL/DISTINCT
+            if let Some(all_distinct) = &self.all_distinct {
+                result.push_str(&all_distinct.render(depth + 1)?);
+            }
+
+            // 各引数の描画
+            {
+                // ALL/DISTINCT、ORDER BYがある場合はインデントを1つ深くする
+                let depth = if self.all_distinct.is_some() || self.order_by.is_some() {
+                    depth + 1
+                } else {
+                    depth
+                };
+
+                // 最初の行のインデント
+                result.extend(repeat_n('\t', depth + 1));
+
+                // 各要素間の改行、カンマ、インデント
+                let mut separator = "\n".to_string();
+                separator.extend(repeat_n('\t', depth));
+                separator.push_str(",\t");
+
+                // Vec<AlignedExpr> -> Vec<&AlignedExpr>
+                let aligned_exprs = self.exprs.iter().collect_vec();
+                let align_info = AlignInfo::from(aligned_exprs);
+
+                result.push_str(
+                    &self
+                        .exprs
+                        .iter()
+                        .map(|a| a.render_align(depth + 1, &align_info))
+                        .collect::<Result<Vec<_>, _>>()?
+                        .join(&separator),
+                );
+            }
+
+            // ORDER BY
+            if let Some(order_by) = &self.order_by {
+                result.push('\n');
+                result.push_str(&order_by.render(depth + 1)?);
+            } else {
+                result.push('\n');
+            }
+
+            result.extend(repeat_n('\t', depth));
+            result.push(')');
+        } else {
+            // 単一行で描画する
+            // ALL/DISTINCT、ORDER BYがある場合は複数行で描画するのでこの分岐には到達しない
+            result.push('(');
+            result.push_str(
+                &self
+                    .exprs
+                    .iter()
+                    .filter_map(|e| e.render(depth + 1).ok())
```

Comment:
filter_mapだとrenderに失敗した要素がスキップされてしまうと思います。  
いずれかの要素でrenderに失敗した場合は、その失敗は親に伝播したいので以下のように直せませんか。

```suggestion
                    .map(|e| e.render(depth + 1))
                    .collect::<Result<Vec<_>, _>>()?
```

Reply:
承知しました。
他にも同様の実装をしている箇所があったので修正を行いました。
https://github.com/future-architect/uroborosql-fmt/pull/12/commits/aaa8bdd520b54c0ebabc3a0344d51eabbec7e176